### PR TITLE
Add matching DObjForeachAnim and nonmatching JObjForeachAnim

### PR DESCRIPTION
**Pull request by r-burns**

### CODE REVIEW

File: src/sysdolphin/baselib/aobj.c [hunk 1]
```diff
@@ -309,112 +310,79 @@ void RObjForeachAnim(HSD_RObj* robj, s32 flags, void* r5, s32 r6, void* r7)
...
+}
+
+// https://decomp.me/scratch/AOTU9
+#ifdef NON_MATCHING
+void JObjForeachAnim(HSD_JObj* jobj, s32 flags, void* r5, s32 r6, void* r7)
+{
+    HSD_JObj* child;
+    HSD_JObj* new_var;
+
+    !(jobj == NULL) ? (void)0 : __assert("aobj.c", 0x2CB, "obj");
```
Comment (PsiLupan) [inline src/sysdolphin/baselib/aobj.c:365]: Can you just do `!=` here? Written this way is even more confusing.

File: src/sysdolphin/baselib/aobj.c [hunk 2]
```diff
@@ -309,112 +310,79 @@ void RObjForeachAnim(HSD_RObj* robj, s32 flags, void* r5, s32 r6, void* r7)
...
+    if (pobj == NULL)
+        return;
+
+    if ((pobj->flags & 0x3000) == 0x1000
+        && pobj->u.x14_unk != NULL && pobj->u.x14_unk->aobj != NULL) {
+        callbackForeachFunc(pobj->u.x14_unk->aobj, pobj, POBJ_TYPE, r5, r6, r7);
+    }
+}
+
+void MObjForeachAnim(HSD_MObj* mobj, s32 flags, void* r5, s32 r6, void* r7)
```
Comment (PsiLupan) [inline src/sysdolphin/baselib/aobj.c:330]: `void PObjForeachAnim(HSD_PObj *obj, HSD_TypeMask mask, void (*func)(void), AObj_Arg_Type arg_type, callbackArg *arg);`

Comment (PsiLupan) [reply src/sysdolphin/baselib/aobj.c:330]: `typedef enum _HSD_TypeMask
{
	AOBJ_BIT      = 1 << 0,
	COBJ_BIT      = 1 << 1,
	DOBJ_BIT      = 1 << 2,
	FOBJ_BIT      = 1 << 3,
	FOG_BIT       = 1 << 4,
	JOBJ_BIT      = 1 << 5,
	LOBJ_BIT      = 1 << 6,
	MOBJ_BIT      = 1 << 7,
	POBJ_BIT      = 1 << 8,
	ROBJ_BIT      = 1 << 9,
	TOBJ_BIT      = 1 << 10,
	WOBJ_BIT      = 1 << 11,
	RENDER_BIT    = 1 << 12,
	CHAN_BIT      = 1 << 13,
	TEVREG_BIT    = 1 << 14,
	CBOBJ_BIT     = 1 << 15,
	ALL_TYPE_BITS = 1 << 16
} HSD_TypeMask;
`

Comment (r-burns) [reply src/sysdolphin/baselib/aobj.c:330]: Should that enum def go in the aobj.h header?

Comment (r-burns) [reply src/sysdolphin/baselib/aobj.c]: Oh, this is very similar to HSD_Type, so I'll put it next to that enum in object.h.

**Comment by r-burns** [comment]
Thanks Psi, that clears it up a lot!

File: include/sysdolphin/baselib/object.h [hunk 3]
```diff
@@ -30,6 +30,26 @@ typedef enum _HSD_Type {
     HSD_MAX_TYPE,
 } HSD_Type;
 
+typedef enum _HSD_TypeMask {
```
Comment (ribbanya) [inline include/sysdolphin/baselib/object.h:33]: Would rather see a macro here
```c
#define MASK_OF(ordinal) (1 << (ordinal-1))
// ...
AOBJ_BITS = MASK_OF(AOBJ_TYPE)
```
and then probably rename `*_BITS` to `*_MASK` to match `HSD_TypeMask`

**Review by ribbanya — CHANGES_REQUESTED** [review]

**Review by ribbanya — APPROVED** [review]

**Review by ribbanya — COMMENTED** [review]
Not super a fan of this but it's fine I guess.
```c
typedef enum _HSD_Type {
    AOBJ_TYPE = 1,
    COBJ_TYPE = 2,
    DOBJ_TYPE = 3,
    FOBJ_TYPE = 4,
    FOG_TYPE = 5,
    JOBJ_TYPE = 6,
    LOBJ_TYPE = 7,
    MOBJ_TYPE = 8,
    POBJ_TYPE = 9,
    ROBJ_TYPE = 10,
    TOBJ_TYPE = 11,
    WOBJ_TYPE = 12,
    RENDER_TYPE = 13,
    CHAN_TYPE = 14,
    TEVREG_TYPE = 15,
    CBOBJ_TYPE = 16,
    HSD_MAX_TYPE,
} HSD_Type;
```

Would rather just set the first one to 1, or create an unused 0 value like `NONE_TYPE`.

**Pull request merged**